sections/main-product.liquid

```diff
@@ -106,6 +106,53 @@
+                id="Inventory-{{ section.id }}"
+                role="status"
+              >
+                {%- if product.selected_or_first_available_variant.inventory_management == 'shopify' -%}
+                  {%- if product.selected_or_first_available_variant.inventory_quantity != 0 -%}
+                    {%- if product.selected_or_first_available_variant.inventory_quantity <= block.settings.inventory_threshold -%}
+                      <svg width="15" height="15" role="presentation">
+                        <circle cx="7.5" cy="7.5" r="7.5" fill="rgb(238,148,65, 0.3)"/>
+                        <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="rgb(238,148,65)"/>
+                      </svg>
+                      {%- if block.settings.show_inventory_quantity -%}
+                        {{- 'products.product.inventory_low_stock_show_count' | t: quantity: product.selected_or_first_available_variant.inventory_quantity -}}
+                      {%- else -%}
+                        {{- 'products.product.inventory_low_stock' | t -}}
+                      {%- endif -%}
+                    {%- else -%}
+                      <svg width="15" height="15" role="presentation">
+                        <circle cx="7.5" cy="7.5" r="7.5" fill="rgb(62,214,96, 0.3)"/>
+                        <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="rgb(62,214,96)"/>
+                      </svg>
+                      {%- if block.settings.show_inventory_quantity -%}
+                        {{- 'products.product.inventory_in_stock_show_count' | t: quantity: product.selected_or_first_available_variant.inventory_quantity -}}
+                      {%- else -%}
+                          {{- 'products.product.inventory_in_stock' | t -}}
+                      {%- endif -%}
+                    {%- endif -%}
+                  {%- else -%}
+                    {%- if product.selected_or_first_available_variant.inventory_policy == 'continue' -%}
+                      <svg width="15" height="15" role="presentation">
+                        <circle cx="7.5" cy="7.5" r="7.5" fill="rgb(62,214,96, 0.3)"/>
+                        <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="rgb(62,214,96)"/>
+                      </svg>
+                      {{- 'products.product.inventory_in_stock' | t -}}
+                    {%- else -%}
+                      <svg width="15" height="15" role="presentation">
+                        <circle cx="7.5" cy="7.5" r="7.5" fill="rgb(200,200,200, 0.3)"/>
+                        <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="rgb(200,200,200)"/>
+                      </svg>
+                      {{- 'products.product.inventory_out_of_stock' | t -}}
+                    {%- endif -%}
+                  {%- endif -%}
+                {%- endif -%}
```

Comment:
Continuing with my earlier comment and looking at the code a bit more I realise there is a lot of repeated code here. Mainly the svg code but this in general hard to read and understand. I also think there is a bug in the current implementation. If you have a product with negative inventory you will still go into the in-stock or low-stock condition, because`inventory_quantity != 0` will match negative numbers and you’ll never reach the ‘continue selling’ condition (in stock), unless the inventory is at exactly 0.

Below is an example of how this _could_ look like if we move the conditional logic into a liquid tag:
<sub>Note: I understand this is few lines longer, but personally I feel this is more readable. Also, I have not run this</sub>
```liquid
{%- if product.selected_or_first_available_variant.inventory_management == 'shopify' -%}
  {%- liquid
    assing variant = product.selected_or_first_available_variant
    assign inventory_quantity = variant.inventory_quantity
    assign show_inventory_quantity = block.settings.show_inventory_quantity
    assign inventory_threshold = block.settings.inventory_threshold
    assign in_stock = false
    assign low_stock = false
    assign continue_selling = false
    assign color_in_stock_rgb = '62, 214, 96'
    assign color_low_stock_rgb = '238, 148, 65'
    assign color_sold_out_rgb = '200, 200, 200'
    assign color = ''
    assign text = ''
    
    if inventory_quantity > 0
      assign in_stock = true
      if inventory_quantity <= inventory_threshold
        assign low_stock = true
      endif
    endif

    if variant.inventory_policy == 'continue'
      assign continue_selling = true
    endif

    if in_stock
      assign color = color_in_stock_rgb
      assign text = 'products.product.inventory_in_stock' | t
      if low_stock
        assign color = color_low_stock_rgb
        if show_inventory_quantity
          assign text = 'products.product.inventory_low_stock_show_count' | t: quantity: inventory_quantity
        else
          assign text = 'products.product.inventory_low_stock' | t
        endif
      elsif show_inventory_quantity
        assign text = 'products.product.inventory_in_stock_show_count' | t: quantity: inventory_quantity
      endif
    elsif continue_selling
      assign color = color_in_stock_rgb
      assign text = 'products.product.inventory_in_stock' | t
    else
      assign color = '200, 200, 200'
      assign text = 'products.product.inventory_out_of_stock' | t
    endif
  -%}
  <svg width="15" height="15" role="presentation">
    <circle cx="7.5" cy="7.5" r="7.5" fill="rgb({{- color -}}, 0.3)"/>
    <circle cx="7.5" cy="7.5" r="5" stroke="rgb(255, 255, 255)" stroke-width="1" fill="rgb({{- color -}})"/>
  </svg>
  {{- text -}}
{%- endif -%}
```

Reply:
Sounds good 😊 glad to help!